zerver/lib/actions.py

```diff
@@ -4472,6 +4493,27 @@ def do_update_message(user_profile: UserProfile, message: Message,
         )
         changed_messages += messages_list
 
+        if new_stream is not None:
+            assert stream_being_edited is not None
+            UserMessage.objects.filter(
+                user_profile__in=[sub.user_profile for sub in guest_subs],
+                message__in=changed_messages,
+            ).delete()
+
+            for msg in changed_messages:
+                delete_event: Dict[str, Any] = {
+                    'type': 'delete_message',
+                    'sender': msg.sender.email,
+                    'sender_id': msg.sender_id,
+                    'message_id': msg.id,
+                    'message_type': 'stream',
+                    'stream_id': stream_being_edited.id,
+                    'topic': orig_topic_name,
+                }
+                delete_event_notify_user_ids = [user.user_profile_id for user in guest_subs]
+                delete_event_notify_users = list(map(subscriber_info, delete_event_notify_user_ids))
+                send_event(user_profile.realm, delete_event, delete_event_notify_users)
```

Comment:
Hrm, this seems really inefficient.  Ideally, we'd send a bulk `delete_messages` event of some sort.  Let me start a thread on #mobile-team about the API considerations here.

Reply:
https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/message_deleted.20events/near/899895

Reply:
Also, I wonder if we could just call a subroutine extracted from `do_delete_messages` for this rather than duplicating the logic.  Ultimately, I think we want to be doing the exact same thing with events, just replacing the query `UserMessage.objects.filter(message_id__in=message_ids)` with `UserMessage.objects.filter(user_profile__is_guest=True, message_id__in=message_ids)` , right?

Reply:
No, we also need old and new stream subscribers list which is already present in "do_update_messages" code.

Reply:
I see, OK.